File: esoteric/3653-xor-after-range-multiplication-queries-i.py
class Solution:
    def xorAfterQueries(self, nums: List[int], queries: List[List[int]]) -> int:
        return reduce(xor, reduce(lambda ls, q: reduce(lambda ls, i: (ls.__setitem__(i, (ls[i] * q[3]) % (10**9 + 7)), ls)[-1], range(q[0], q[1]+1, q[2]), ls), queries, nums))

File: elegant/3653-xor-after-range-multiplication-queries-i.cpp
class Solution {
public:
    int xorAfterQueries(vector<int>& nums, vector<vector<int>>& queries)
    {
        for (const span q : queries) {
            const int v { q[3] };
            for (const int i : views::iota(q[0], q[1] + 1) | views::stride(q[2])) {
                nums[i] = (1LL * nums[i] * v) % 1'000'000'007;
            }
        }

        return ranges::fold_left(nums, 0, bit_xor<int>{});
    }
};
